plugins/configuration.rb

```diff
@@ -0,0 +1,114 @@
+module Jekyll
+  class ConfigurationBlock < Liquid::Block
+    TYPE_LINKS = {
+      'action'       => '/docs/scripts/',
+      'device_class' => '/components/%{component}/#device_class',
+      'template'     => '/docs/configuration/templating/',
+    }
+
+    def initialize(tag_name, text, tokens)
+      super
+      @component, @platform = text.strip.split('.', 2)
+    end
+
+    def slug(key)
+      key.downcase.strip.gsub(' ', '-').gsub(/[^\w\-]/, '')
+    end
+
+    def type_class(type)
+      ((type.is_a? Array) ? type.join(' ') : type).downcase
```

Comment:
Funny that you're wrapping your `type.is_a?` expression in parentheses because you are omitting the parentheses on the function params (`type.is_a?(Array)`)

Reply:
:joy: This is my first Ruby project. I still don't know when to, and when not to, use parenthesis!

Reply:
I usually use them always as I like consistency. Not using them can get you into situations like this.

Reply:
Good to know. I will stick to that as well. I've always been confused (I think since Visual Basic 3!) why it is sometimes optional to use parenthesis.